flocker/control/httpapi.py

```diff
@@ -812,11 +818,15 @@ def replace_configuration(self, applications, deployment):
         :param deployment: Configuration of which applications run on
             which nodes.
         """
+        # XXX we're going to have to deal with UUIDs better,
+        # somehow... look up state of cluster? Change deployment
+        # configuration format?
         try:
             configuration = FigConfiguration(applications)
             if not configuration.is_valid_format():
                 configuration = FlockerConfiguration(applications)
             return self.persistence_service.save(model_from_configuration(
+                self.cluster_state_service.as_deployment(),
```

Comment:
Would be easier to read if this was a keyword argument....to make it clear that we're passing a `DeploymentState` not `Deployment` ...despite the method name `as_deployment`.